### ISSUE TITLE
`helpful-string` false negative on dataclasses with `repr=False`

### ISSUE THREAD

**Issue by DetachHead** [KotlinIsland/basedmypy#748]
```py
from dataclasses import dataclass


@dataclass(repr=False)
class Foo:
    a: int = 1


print(f"{Foo(1)}")  # <__main__.Foo object at 0x000002156543C530>
```
